crud-app/app/Http/Controllers/ProductsGet.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use \App\Models\Product;
+
+class ProductsGet extends Controller
+{
+    /**
+     * Handle the incoming request.
+     */
+    public function __invoke(Request $request)
+    {
+        try{
+
+           $sortBy = $request->query('sortBy') ?? 'name';
+           $direction =  $request->query('direction') ?? 'asc';
+           $products = Product::orderBy($sortBy, $direction)->get();
+           return response()->json(['products' => $products], 200);  
+    } catch (\Exception $e) {
+        return response()->json(['message' => 'Error retrieving products: ' . $e->getMessage()], 400);
```

Comment:
Don't use $e->getMessage(). This error message is intended for consumers of your API. You don't want to expose them to Mysql errors that are internal to your application. Update and leave generic for now.

Reply:
Understood, fixed!